File: Ecity/NetWorkManager/SubNetWorkManager.m
//
//  SubNetWorkManager.m
//  Ecity
//
//  Created by wangwenbing on 2017/4/25.
//  Copyright © 2017年 Centit. All rights reserved.
//

#import "SubNetWorkManager.h"

@interface SubNetWorkManager ()

@end

#define projectPath   @"service/project/projectsearch/listprojectinfo"

@implementation SubNetWorkManager
    
+ (instancetype)sharedInstanceSession
{
    static SubNetWorkManager *subNetWorkManager = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSURLSessionConfiguration *defaultsConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
        subNetWorkManager = [[self alloc] initWithBaseURL:nil sessionConfiguration:defaultsConfiguration];
        [self reachabilityManager];
    });
    return subNetWorkManager;
}
    
- (AnyPromise *)projectList
{
    NSString  *urlString = nil;
    [self.requestSerializer setTimeoutInterval:10];
    
    urlString = [ServerAndPort stringByAppendingString:projectPath];
   
    urlString = [urlString stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
    
    AnyPromise *promise = [self pmk_GET:urlString parameters:@{}].then(^(OVCResponse *response){
        
        return response;
        
    }).catch(^(NSError *error) {
        
        return error;
        
    });
    
    return promise;
}
@end

File: Ecity/NetWorkManager/SubNetWorkManager.swift
//
//  SubNetWorkManager.swift
//  Ecity
//
//  Created by wangwenbing on 2017/4/26.
//  Copyright © 2017年 Centit. All rights reserved.
//

import UIKit

class SubNetWorkManager: BaseNetWorkManager, Ecity_Protocol {
    
    final class FirstRunManager: NSObject {
        static let sharedInstance = FirstRunManager()
        private override init() {
        }
        
    }
   
    
    final class func sharedInstanceSession() -> SubNetWorkManager
        
    {
        
        var sharedInstance: SubNetWorkManager? = nil
        
        if sharedInstance == nil {
            let sessionConfig = URLSessionConfiguration.default 
            sharedInstance = BaseNetWorkManager.init(baseURL: nil, sessionConfiguration: sessionConfig) as? SubNetWorkManager
            sharedInstance!.reachabilityManager = SubNetWorkManager.reachabilityManager()
        }
      
        
        return sharedInstance!
        
        
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    func projectList() -> AnyPromise {
        
    
        
        var urlString: String = ""
        
        urlString = String.init(format: "%@/%@", arguments: [ServerAndPort,ProjectPath])
        urlString = String.init(describing: NSCharacterSet.urlQueryAllowed)
        
        let anyPromise = self.pmk_GET(urlString, parameters: nil).then { (response) -> AnyPromise in
            
            let promise = AnyPromise.promiseWithValue(response)
            
            return promise
            
        }.catch { (error) in
            
        }
        
        return AnyPromise(anyPromise)
    }
    
}
